tox.ini

```diff
@@ -0,0 +1,92 @@
+[tox]
+envlist = py{39,310,311,312,313,314,py310}
+labels =
+    check = typing, gen_exports, type_completeness, pip_compile
+    cython = py39-cython2,py39-cython,py311-cython2,py313-cython
+
+# TODO:
+# * environment to check coverage
+# * replace ci.sh
+#   * --verbose --durations=10
+#   * -p _trio_check_attrs_aliases
+#   * mypy cache
+#   * LSP
+#   * apport
+# * use tox in CI
+# * switch to nox?
+# * move to pyproject.toml?
+#   * this means conditional deps need to be replaced
+
+# protip: install uv-tox for faster venv generation
+
+[testenv]
+# use wheels instead of sdist, significantly faster install
+package = wheel
+wheel_build_env = .pkg
+deps =
+    hypothesis: hypothesis
+    -r test-requirements.txt
+set_env =
+    slow: TOX_RUN_SLOW = '--run-slow'
+commands =
+    pytest {env:TOX_RUN_SLOW:} {posargs}
+
+[testenv:no_test_requirements]
+commands =
+    pytest --skip-optional-imports {posargs}
+
+[testenv:docs]
+deps =
+    -r docs-requirements.txt
+# base_python synced with .readthedocs.yml
+base_python = 3.11
+commands =
+    sphinx-build {posargs:--fresh-env} docs/source docs/build
+
+[testenv:py39-cython2,py39-cython,py311-cython2,py313-cython]
+deps =
+    cython
+    cython2: cython<3
+    setuptools ; python_version >= '3.12'
+commands =
+    cythonize --inplace -X linetrace=True tests/cython/test_cython.pyx
+    python --version
+    cython --version
+    python -m tests.cython.run_test_cython
+
+[testenv:gen_exports]
+deps =
+    -r test-requirements.txt
+base_python = 3.13
+commands =
+    python ./src/trio/_tools/gen_exports.py --test
+
+[testenv:pip_compile]
+base_python = 3.13
+commands =
+    pre-commit run pip-compile --all-files
+
+[testenv:typing]
+deps =
+    -r test-requirements.txt
+    exceptiongroup
+base_python = 3.13
+set_env =
+    PYRIGHT_PYTHON_IGNORE_WARNINGS=1
+commands =
+    mypy --platform linux
+    mypy --platform darwin
+    mypy --platform win32
```

Comment:
Shouldn't this use `mypy_annotate`?

Reply:
hm. Maybe, but `mypy_annotate` is less needed with tox supressing output of succesful commands and in general reformatting the output. One of the problems with `check.sh` was simply finding what it was that failed, which I think is what prompted the creation of it. But we don't want it when running outside of CI, so needs some logic to differentiate.
But if we don't use it here then it should just be removed.

I'll procrastinate on it a little bit, and see what the output looks like in practice. I suspect the end result is going to be removing `mypy_annotate`